### PR TITLE
Feature/16 auth

### PR DESCRIPTION
**Pull request by safyurtago**
Create Auth Part
Create UserGuard and CookieGetter Decorators

### CODE REVIEW

File: src/auth/auth.service.ts [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+    const tokens = await this.getTokens(newUser.id, newUser.email, newUser.role);
+    const hashedRefreshToken = await bcrypt.hash(tokens.refresh_token, 12);
+    const activationLink = v4();
+
+    await this.prismaService.user.update({
+      data: { hashedRefreshToken, activationLink },
+      where: { id: newUser.id },
+    });
+
+    res.cookie('refresh_token', tokens.refresh_token, {
```
Comment (bakhromov96) [inline src/auth/auth.service.ts:44]: Move the setting cookie to some `private setRefreshTokenCookie` function to reduce code redundancy.
change hardcoded maxAge to some constant value.

File: src/auth/auth.service.ts [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+      maxAge: 7 * 24 * 60 * 60 * 1000,
+      httpOnly: true,
+    });
+    return tokens;
+  }
+  async login(loginDto: LoginDto, res: Response) {
+    const { email, password } = loginDto;
+    const user = await this.prismaService.user.findUnique({ where: { email } });
+    if (!user) throw new UnauthorizedException('User not found');
```
Comment (bakhromov96) [inline src/auth/auth.service.ts:53]: move all this validation to a helper function like `validateUser`

**Review by bakhromov96 — CHANGES_REQUESTED** [review]

File: src/auth/auth.service.ts (hunk 2 above)
Comment (safyurtago) [reply src/auth/auth.service.ts]: I updated all you said, Can you review one more time please?

**Review by bakhromov96 — APPROVED** [review]

**Review by mirsadikov — APPROVED** [review]
All good

**Pull request merged**